R/zchunk_L2072.ag_water_irr_mgmt.R

```diff
@@ -1,67 +1,333 @@
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(basin_to_country_mapping[c("GLU_code", "GLU_name")], by = c("GLU" = "GLU_code")) %>%
+      # Copy to both high and low management levels
+      repeat_add_columns(tibble::tibble(MGMT = c("hi", "lo"))) %>%
+      # Add sector, subsector, technology names
+      mutate(IRR_RFD = "IRR",
+             AgSupplySector = GCAM_commodity,
+             AgSupplySubsector = paste(GCAM_commodity, GLU_name, sep = "_"),
+             AgProductionTechnology = paste(GCAM_commodity, GLU_name, IRR_RFD, MGMT, sep = "_"),
+             water_type = "water consumption",
+             coefficient = round(BlueIrr_m3kg, aglu.DIGITS_CALOUTPUT)) %>%
+      filter(coefficient > 0) %>%
+      # Assume coefs stay constant, copy to all model years
+      repeat_add_columns(tibble::tibble(year = MODEL_YEARS)) %>%
+      # Standardize irrigation water input names
+      mutate(water_sector = "Irrigation",
+             minicam.energy.input = set_water_input_name(water_sector, water_type, A03.sector, GLU = GLU_name)) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["AgCoef"]])) ->
+      L2072.AgCoef_IrrWaterCons_ag_mgmt
+
+    # Compute biophysical water consumption IO coefficients (km3/Mt crop = m3/kg) by region / irrigated crop / year / GLU / management level
+    L165.TotIrr_m3kg_R_C_GLU %>%
+      ungroup() %>%
+      # Initial adjustment: drop any water coefs for region/crop/GLU combinations that aren't written out
+      semi_join(L161.ag_irrProd_Mt_R_C_Y_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU")) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(basin_to_country_mapping[c("GLU_code", "GLU_name")], by = c("GLU" = "GLU_code")) %>%
+      # Copy to both high and low management levels
+      repeat_add_columns(tibble::tibble(MGMT = c("hi", "lo"))) %>%
+      # Add sector, subsector, technology names
+      mutate(IRR_RFD = "IRR",
+             AgSupplySector = GCAM_commodity,
+             AgSupplySubsector = paste(GCAM_commodity, GLU_name, sep = "_"),
+             AgProductionTechnology = paste(GCAM_commodity, GLU_name, IRR_RFD, MGMT, sep = "_"),
+             water_type = "biophysical water consumption",
+             coefficient = round(TotIrr_m3kg, aglu.DIGITS_CALOUTPUT)) %>%
+      filter(coefficient > 0) %>%
+      # Assume coefs stay constant, copy to all model years
+      repeat_add_columns(tibble::tibble(year = MODEL_YEARS)) %>%
+      # Standardize irrigation water input names
+      mutate(water_sector = "Irrigation",
+             minicam.energy.input = set_water_input_name(water_sector, water_type, A03.sector, GLU = GLU_name)) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["AgCoef"]])) ->
+      L2072.AgCoef_IrrBphysWater_ag_mgmt
+
+    # Compute biophysical water consumption IO coefficients (km3/Mt crop = m3/kg) by region / rainfed crop / year / GLU / management level
```

Comment:
Seems like the exact same logic is happening here three times in a row? If so let's create an internal function to handle this.